subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java

```diff
@@ -38,6 +39,16 @@ public interface ProjectLayout {
      */
     Directory getProjectDirectory();
 
+    /**
+     * Returns the settings directory.
+     * <p>
+     * The settings directory is the directory containing the settings file.
+     *
+     * @since 8.13
+     */
+    @Incubating
+    Directory getSettingsDirectory();
```

Comment:
💭  I wonder about the placement of the getter relative to other functions. Right now it's in the middle between the project-specific stuff.

It'd be best if we move it on the side. Though, putting it first does not seem right, as the interface should be primarily about the project. I'd vote for adding it to the end of the file